programming_201.rst

```diff
@@ -86,10 +88,33 @@ You will see this for Ruby:
   Hello again
   => 1..3
 
-Ruby is a very good tool to write scripts.
+And using loop variables inside the loop, again in **C++** first:
+
+.. code-block:: cpp
+
+  for(int i = 0; i < 3; i++) {
+      if( i == 2 ){
+          std::cout << "This is loop " << i.str() << "!"
+      }
+  }
+  
+Which is much simpler in Ruby:
+.. code-block:: ruby
+
+  > (1..3).each { |num| puts 'This is loop #{num}!' if num.eql? 3 }
```

Comment:
This is not exactly the best loop, as it uses a Range, and the quotations are incorrect for string interpolation.

How about these two, which are a little more Rubyish?

```ruby
> 4.times { |num| puts "This is loop #{num}!" if num.eql? 3 }
This is loop 3!
=> 4
 1.upto(4) { |num| puts "This is loop #{num}!" if num.eql? 3 }
This is loop 3!
=> 1
```